Change Simbad query_objectids_async to yield meaningful response content

kakirastern opened the pull request:
Fixes #1413. 

Now when I do the following:
```
>>> from astroquery.simbad import Simbad
>>> response = Simbad.query_objectids_async("m1")
```
Instead of getting the rather bland `<Response [200]>` when I do `print(response)`, now I get instead:
```
::script::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

format object "%IDLIST"
query id m1

::console:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

C.D.S.  -  SIMBAD4 rel 1.7  -  2019.04.12CEST09:39:15
total execution time: 0.115 secs
simbatch done

::data::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

3FHL J0534.5+2201
SNR G184.6-05.8
1H 0531+219
1M 0531+219
2C  481
2E  1309
2U 0531+22
3A 0531+219
3C 144
3C 144.0
3CR 144
3U 0531+21
4C 21.19
4U 0531+21
AJG   1
CTA  36
CTB 18
Cul 0531+21
Cul 0531+219
DA 179
DB  38
H 0534+21
H 0531+219
IRAS 05314+2200
LBN   833
LBN 184.62-05.65
M   1
Mills 05+2A
NAME CRAB NEB
NAME Crab
NAME Crab Nebula
NAME Tau A
NAME Taurus A
NGC  1952
NRAO 214
NRL  2
PKS 0531+219
SH  2-244
SIM 0531+21.0
VRO 21.05.01
X Tau X-1
X Tau XR-1
[BM83] X0531+219
[DGW65]  25
[PT56]  5
1ES 0532+21.5
2E 0531.5+2159
PBC J0534.5+2201
SWIFT J0534.6+2204
SWIFT J0534.5+2200
GRS G184.60 -05.80
W 9
NVSS J053428+220202
ARGO J0535+2203
3FGL J0534.5+2201i
TeV J0534+220
```

kakirastern commented:
But then I ran into some troubles with the tests. So I checked again with at least one other module, namely the `SDSS` module... Then, when I do the following:
```
>>> from astroquery.sdss import SDSS
>>> import numpy as np
>>> ra = np.array([220.064728084,220.064728467,220.06473483])
>>> dec = np.array([0.870131920218,0.87013210119,0.870138329659])
>>> from astropy.coordinates import SkyCoord
>>> coordinates = SkyCoord(ra, dec, frame='icrs', unit='deg')
>>> response = SDSS.query_crossid_async(coordinates)
```
Then, I would get:
```
>>> print(response)
<Response [200]>
```
But then I can do `response.content` or `response.text` to get the payload... Like the following:
```
>>> response.text
'#Table1\nobj_id,objID,ra,dec,psfMag_u,psfMagerr_u,psfMag_g,psfMagerr_g,psfMag_r,psfMagerr_r,psfMag_i,psfMagerr_i,psfMag_z,psfMagerr_z,obj_id1,type\nobj_0,1237648705666875633,220.064728466608,0.870132101190482,22.03464,0.1391412,19.27899,0.01703184,17.90074,0.01622461,17.22311,0.01888538,16.86472,0.01686881,1237648705666875633,STAR\nobj_1,1237648705666875633,220.064728466608,0.870132101190482,22.03464,0.1391412,19.27899,0.01703184,17.90074,0.01622461,17.22311,0.01888538,16.86472,0.01686881,1237648705666875633,STAR\nobj_2,1237648705666875633,220.064728466608,0.870132101190482,22.03464,0.1391412,19.27899,0.01703184,17.90074,0.01622461,17.22311,0.01888538,16.86472,0.01686881,1237648705666875633,STAR\n'
```

kakirastern commented:
Therefore, if the idea stated in Issue #1413 is to be followed then some tests might need to be rewritten... Will follow up on Slack to check if this is the direction to go. Otherwise, given the current setup before any changes introduced by the PR, normally if we do:
```
>>> from astroquery.simbad import Simbad
>>> response = Simbad.query_objectids_async("Polaris")
>>> print(response)
<Response [200]>
```
But if you follow up with `print(response.text)` you would get 
```
::script::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

format object "%IDLIST"
query id Polaris

::console:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

C.D.S.  -  SIMBAD4 rel 1.7  -  2019.04.12CEST10:13:05
total execution time: 0.031 secs
simbatch done

::data::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

NAME Lodestar
PLX  299
SBC9 76
*   1 UMi
* alf UMi
AAVSO 0122+88
ADS  1477 A
AG+89    4
BD+88     8
CCDM J02319+8915A
CSI+88     8  1
FK5  907
GC  2243
GCRV  1037
GEN# +1.00008890A
GSC 04628-00237
HD   8890
HIC  11767
HIP  11767
HR   424
IDS 01226+8846 A
IRAS 01490+8901
JP11   498
N30  381
NAME Polaris
PMC 90-93   640
PPM    431
ROT  3491
SAO    308
SBC7    51
SKY#  3738
TD1   835
TYC 4628-237-1
UBV   21589
UBV M   8201
V* alf UMi
PLX  299.00
WDS J02318+8916Aa,Ab
ADS  1477 AP
** WRH   39
WDS J02318+8916A
** STF   93A
2MASS J02314822+8915503
NAME North Star
WEB  2438
Gaia DR2 576402619921505664
```
which as is expected.

kakirastern commented:
I can rebase and drop the last commit and then rewrite the relevant tests if necessary, given that's indeed the desirable option.

pep8speaks commented:
Hello @kakirastern! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`astroquery/simbad/core.py`](https://github.com/astropy/astroquery/blob/9b035ab78bbca908ee36eadc64ed52fb55eba114/astroquery/simbad/core.py):

> [Line 892:1](https://github.com/astropy/astroquery/blob/9b035ab78bbca908ee36eadc64ed52fb55eba114/astroquery/simbad/core.py#L892): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

kakirastern commented:
PR is ready for a review.

bsipocz commented:
@kakirastern - this is not exactly the solution I had in mind. Are you planning to come back to this issue, or shall we open another PR for it?

kakirastern commented:
Hi @bsipocz. I plan to come back to this PR, but will need some more details regarding how to change it so that the PR will conform to your expectations. I probably misunderstood things the first time around. My apologies for that.

bsipocz commented:
No worries, misunderstanding happens. Basically what is needed here is that line `[8]` and `[9]` work like lines `[10]` and `[11]`.
Line `[7]` is the correct behaviour. So have a look at how this is done in `query_object()` and `query_object_async()`. 


```
In [5]: from astroquery.simbad import Simbad

In [6]: simbad = Simbad()

In [7]: simbad.query_objectids_async('m1')
Out[7]: <Response [200]>

In [8]: simbad.query_objectids_async('m1', get_query_payload=True)
Out[8]: <Response [200]>

In [9]: simbad.query_objectids('m1', get_query_payload=True)
Out[9]: <Response [200]>

In [10]: simbad.query_object('m1', get_query_payload=True)
Out[10]: {'script': '\nvotable {main_id,coordinates}\nvotable open\nquery id  m1  \nvotable close'}

In [11]: simbad.query_object_async('m1', get_query_payload=True)
Out[11]: {'script': '\nvotable {main_id,coordinates}\nvotable open\nquery id  m1  \nvotable close'}
```

kakirastern commented:
I have modified the code so that when `get_query_payload` is `True`, then the `request_payload` is returned instead of `response`, after I have rebased the PR. Now, I get the following:

```
>>> from astroquery.simbad import Simbad
>>> simbad = Simbad()

>>> simbad.query_objectids_async('m1')
<Response [200]>

>>> simbad.query_objectids_async('m1', get_query_payload=True)
{'script': 'format object "%IDLIST"\nquery id m1'}

>>> simbad.query_objectids('m1', get_query_payload=True)
{'script': 'format object "%IDLIST"\nquery id m1'}
```

Please let me know if anything is amiss or if any further modification is required.

bsipocz commented:
@kakirastern - thank you, this is the one!
